README.md

```diff
@@ -1,52 +1,184 @@
+
+// Get top bots ranking
+const rankings = await client.getRankings({
+  sortBy: 'monthly_votes_rank',
+  sortMethod: 'desc'
+});
+console.log(rankings.data);
 ```
 
-## Methods
+## API Reference
 
-**Returns information on a given bot**
+### Constructor
 
-`dbl.getBot(id)`
+```typescript
+new Client(token: string)
+// or
+new Client({ token: string })
+```
 
-id = ID of the bot
+### Methods
 
-`dbl.getBotHistory(id, limit)`
+#### `getBot(botId: string): Promise<BotData>`
+Fetches detailed information about a specific bot.
 
-**Returns the historical data separated by hour**
+```typescript
+const bot = await client.getBot('583807014896140293');
+```
 
-id = ID of the bot
+#### `getBotHistorical(botId: string, timeFrame: HistoricalDataTimeFrame, type: HistoricalDataTypes): Promise<HistoricalDataResponse>`
+Fetches historical data for a specific bot.
+
+```typescript
+// Available timeframes
+type HistoricalDataTimeFrame = 
+  'alltime' | '5y' | '3y' | '1y' | '90d' | 
+  '30d' | '7d' | '1d' | '12hr' | '6hr';
+
+// Available data types
+type HistoricalDataTypes = 
+  'monthly_votes' | 'total_votes' | 
+  'server_count' | 'shard_count';
+
+const history = await client.getBotHistorical(
+  '583807014896140293',
+  '30d',
+  'monthly_votes'
+);
+```
 
-limit = Amount of data to fetch
+#### `getBotRecent(botId: string): Promise<RecentDataResponse>`
+Fetches recent statistics for a specific bot.
 
-`dbl.getTop(by, limit)`
+```typescript
+const recent = await client.getBotRecent('583807014896140293');
+```
 
-**Fetches leaderboard for given list**
+#### `getRankings(options: RankingsRequest): Promise<RankingsResponse>`
+Fetches bot rankings based on specified criteria.
 
-by = Which list to fetch, options = `servers`, `votes`, `monthly votes`, `shards`
+```typescript
+const rankings = await client.getRankings({
+  sortBy: 'monthly_votes_rank',
+  sortMethod: 'desc',
+  limit: 250 // Optional, defaults to 100
+});
+```
 
-limit = Amount of data to fetch
+### Types
+
+```typescript
+interface BotData {
+  id: string;
+  name: string;
+  server_count: number;
+  monthly_votes: number;
+  total_votes: number;
+  // ... and more
+}
+
+interface RecentDataResponse {
+  hourlyData: RecentData[];
+  dailyData: RecentData[];
+}
+
+interface RankingsResponse {
+  totalBotCount: number;
+  data: RankingsData[];
+}
+```
 
-`dbl.getUsersBots(id)`
+## Error Handling
+
+The client throws typed errors for different scenarios:
+
+```typescript
+try {
+  await client.getBot('invalid-id');
+} catch (error) {
+  if (error.message.includes('Rate limit')) {
+    // Handle rate limit
+    console.log('Rate limited, try again later');
+  } else {
+    // Handle other errors
+    console.error('Error:', error.message);
+  }
+}
+```
 
-**Gives .user as user object and .bots is a list of bots the user owns**
+## Rate Limits
 
-id = ID of the user
+The API has a rate limit of 60 requests per minute. The client will throw an error when the rate limit is exceeded.
```

Comment:
it's worth mentioning that this varies per route.

You can link to the docs [here](https://docs.topstats.gg/authentication/ratelimits/)